### PR TITLE
long polling support

### PR DESCRIPTION
**Pull request by BarAgam**
long polling support

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GGMessaging/pull/2" target="_blank">ReviewNinja

File: routes/common.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+
+
+function handleGetAllMessages(next, res) {
+    message.find(function (err, messages) {
+        if (err) {
+            //forward to error handling
+            next(err);
+        }
+
```
Comment (teamagam3) [inline routes/common.js:15]: Comment is missing here

File: app.js [hunk 2]
```diff
@@ -34,7 +36,20 @@ app.use(bodyParser.urlencoded({extended: false}));
 app.use(cookieParser());
 app.use(express.static(path.join(__dirname, 'public')));
 
-app.use('/messages', routes);
+app.use('/messages', messagesRoute);
+
+app.use('/long/messages', longPollingMessagesRoute);
+
+//timeout catcher
+app.use(function (req, res, next) {
```
Comment (teamagam1) [inline app.js:44]: who is going to first? route to /long/messages or timeout?
does the order in app.js is important?

Comment (BarAgam) [inline app.js:44]: The order is given by the order the order of the app.use methods. that means a request will first try and match /message and /long/messages route and afterwards (if a response is not already sent) it'll be forwarded to the timeout catcher and the error handler

File: EventEmitters/mongoEventsEmitter.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const EventEmitter = require('events');
+const util = require('util');
+
+function MessageCollectionEmitter() {
```
Comment (teamagam1) [inline EventEmitters/mongoEventsEmitter.js:4]: documentation.
who uses those events and what for?

File: routes/long_polling_messages.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+/**
+ * GET all messages with date later than given parameter
+ * limits to K results
+ */
+router.get('/fromDate/:date', function (req, res, next) {
+
+    common.getMessagesFromDate(req, function (err) {
+        next(err);
+    }, function (messages) {
```
Comment (teamagam1) [inline routes/long_polling_messages.js:45]: this is the same code as above.
can't you merge those two to one method?

File: routes/common.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
+/**
+ * Common methods used by routers
+ */
+
+var message = require('../models/message');
+var config = require('../config/config');
+
+
+function getAllMessages(onFailure, onSuccess) {
```
Comment (teamagam1) [inline routes/common.js:9]: documentation.
difference between those two methods

**Comment by teamagam1** [comment]
when i used long pulling i got this message:
timeout reached. removing listener [this is OK]
Error: Can't set headers after they are sent.
.
.
.
[stack trace]
!fix

**Pull request merged**